### ISSUE TITLE
Invalid insertion done with setup time

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#593]
TL;DR something is missing in the #547 implementation

Someone reported an instance hitting [this assertion](https://github.com/VROOM-Project/vroom/blob/cb6ebd40baff5b8dab4ab716ecac0e79e4bc145a/src/structures/vroom/tw_route.cpp#L178). It's a check in `TWRoute::fwd_update_earliest_from`, applied upon a route modification (call from `TWRoute::replace`), that makes sure updated earliest date for tasks are always smaller than latest dates.

After some debugging, I could confirm that this happens in a situation where an *invalid* move is applied, even if the matching call to `TWRoute::is_valid_addition_for_tw` previously returned `true`.

More precisely, we have to check dates validity one step further in the situation where setup time should be applied to the first task *after* the replacement range:

https://github.com/VROOM-Project/vroom/blob/cb6ebd40baff5b8dab4ab716ecac0e79e4bc145a/src/structures/vroom/tw_route.cpp#L771-L789

The overall logic is that a *new* setup time addition for `j_after` invalidates the earliest time for next job, hence the need to check one step further. But here on line 184 `current.earliest + next.travel` is only a *candidate* for `j_after`s earliest date and we miss pushing it later if the first TW for `j_after` starts later.

**Comment by jcoupey** [comment]
The fix is now also available in the `release/1.11` branch (207d8b7 cherry-picked as 71eecc9).
